src/data/shortlist.h

```diff
@@ -60,8 +60,8 @@ class SampledShortlistGenerator : public ShortlistGenerator {
   size_t trgIdx_;
   bool shared_{false};
 
-  std::random_device rd_;
-  std::mt19937 gen_;
+  // static thread_local std::random_device rd_;
+  static thread_local std::unique_ptr<std::mt19937> gen_;
```

Comment:
Have you checked this compiles?

Reply:
Ah, this is guarded by #ifdef/#endif, that explains how that would compile despite being wrong. The interface for review on github is really bad.